internal/fleet/agent_policy_resource.go

```diff
@@ -140,6 +163,25 @@ func resourceAgentPolicyCreate(ctx context.Context, d *schema.ResourceData, meta
 	}
 	req.MonitoringEnabled = &monitoringValues
 
+	if tags, ok := d.GetOk("global_data_tags"); ok {
+		tagMap := tags.(map[string]interface{})
+
+		if len(tagMap) > 0 && serverVersion.LessThan(minVersionGlobalDataTags) {
+			return diag.FromErr(fmt.Errorf("'global_data_tags' is supported only for Elasticsearch v%s and above", minVersionGlobalDataTags.String()))
+		}
+
+		globalDataTags := make([]fleetapi.GlobalDataTag, 0, len(tagMap))
+		for key, value := range tagMap {
+			globalDataTags = append(globalDataTags, fleetapi.GlobalDataTag{
+				Name:  key,
+				Value: value.(string),
+			})
+		}
+		req.GlobalDataTags = globalDataTags
+	} else {
+		req.GlobalDataTags = make([]fleetapi.GlobalDataTag, 0) // Ensure it's an empty array
```

Comment:
Why is it important that this is an empty slice and not just left as nil?

Reply:
When I would create a resource that contains a global_data_tags map and then update it to completely remove the map (not just an item in the map) the API would error out if this was nil, and if it was omitted when empty the settings would remain and just no longer be tracked in state. Same goes for the other comment.

It's possible I could set GlobalDataTags in the AgentPolicyCreateRequest struct to omitempty while keeping AgentPolicyUpdateRequest as is but I was trying to keep the implementation consistent. This also might change once I run the commands to regenerate fleet.gen.go so we'll see